File: backend/src/main/java/info/gamewise/lor/videos/entity/VideoRepository.java
package info.gamewise.lor.videos.entity;

import org.springframework.data.mongodb.repository.MongoRepository;
import org.springframework.data.querydsl.QuerydslPredicateExecutor;

import java.util.List;

interface VideoRepository extends MongoRepository<VideoJpaEntity, String>, QuerydslPredicateExecutor<VideoJpaEntity> {
    boolean existsByVideoId(String videoId);
    List<VideoJpaEntity> findByChannel(String channel);
}

File: backend/src/main/kotlin/info/gamewise/lor/videos/entity/VideoRepository.kt
package info.gamewise.lor.videos.entity

import org.springframework.data.mongodb.repository.MongoRepository
import info.gamewise.lor.videos.entity.VideoJpaEntity
import org.springframework.data.querydsl.QuerydslPredicateExecutor

internal interface VideoRepository : MongoRepository<VideoJpaEntity, String> {
    fun existsByVideoId(videoId: String): Boolean
    fun findByChannel(channel: String): List<VideoJpaEntity>
}